crawler.py

```diff
@@ -189,94 +189,102 @@ def run(website, driver):
     return saved_sequence
 
 
-def run_all(driver, data_folder):
-    set_timeouts(driver)
+def run_all():
+    # Environmental vars set to overwrite default location of .gcda files
+    if sys.platform.startswith('linux') or sys.platform.startswith('darwin'):
+        prefix = '/builds/worker/workspace/build/src/'
+        strip_count = prefix.count('/')
+    elif sys.platform.startswith('cygwin') or sys.platform.startswith('win32'):
+        prefix = 'z:/build/build/src/'
+        strip_count = prefix.count('/') + 1
 
-    if not os.path.exists(data_folder):
-        os.makedirs(data_folder)
+    # Remove a prefix from the path where .gcda files are stored
+    os.environ['GCOV_PREFIX_STRIP'] = str(strip_count)
+    os.environ['PATH'] += os.pathsep + os.path.abspath('tools')
+    os.environ['MOZ_HEADLESS'] = '1'
 
-    with open('websites.txt') as f:
-        for i, website in enumerate(f):
-            if os.path.exists('{}/{}.txt'.format(data_folder, i)):
-                continue
+    # Create temporary directories with context manager
+    with tempfile.TemporaryDirectory() as gcov_dir, tempfile.TemporaryDirectory() as jsvm_dir:
+        os.environ['GCOV_PREFIX'] = gcov_dir
+        os.environ['JS_CODE_COVERAGE_OUTPUT_DIR'] = jsvm_dir
 
-            try:
-                sequence = run(website, driver)
+        with open('websites.txt') as f:
+            for i, website in enumerate(f):
+                driver = webdriver.Firefox(firefox_binary='tools/firefox/firefox-bin')
+                set_timeouts(driver)
 
-                with open('{}/{}.txt'.format(data_folder, i), 'w') as f:
-                    f.write('Website name: ' + website + '\n')
-                    for element in sequence:
-                        f.write(json.dumps(element) + '\n')
+                # All steps are stored in new folder
+                data_folder = str(uuid.uuid4())
 
-            except:  # noqa: E722
-                traceback.print_exc()
-                close_all_windows_except_first(driver)
+                if not os.path.exists(data_folder):
+                    os.makedirs(data_folder)
 
-    driver.quit()
+                if os.path.exists('{}/{}.txt'.format(data_folder, i)):
+                    continue
 
+                try:
+                    sequence = run(website, driver)
+                    with open('{}/{}.txt'.format(data_folder, i), 'w') as f:
+                        f.write('Website name: ' + website + '\n')
+                        for element in sequence:
+                            f.write(json.dumps(element) + '\n')
 
```

Comment:
The try-except should end here, we only want to ignore exceptions coming from Selenium, but not from our own code.